### PR TITLE
DR | Cleanup & fix end-to-end tests

### PR DESCRIPTION
**Pull request by Mottie**
## Summary

- _(Summarize the changes that have been made to the platform)_
  > This PR moves common e2e utils & mock data into `appeals/shared` folder and updates all e2e tests to use common code & data.
- _(If bug, how to reproduce)_
- _(What is the solution, why is this the solution)_
  > Clean up duplicate mock data & utilities
- _(Which team do you work for, does your team own the maintenance of this component?)_
  > Benefits decision reviews
- _(If using a flipper, what is the end date of the flipper being required/success criteria being targeted)_

## Related issue(s)

[#67960](https://github.com/department-of-veterans-affairs/va.gov-team/issues/67960)

## Testing done

- _Describe what the old behavior was prior to the change_
  > 995 e2e test was disallowed, we've set `scrollBehavior: 'nearest'` in the Cypress settings in hopes of making this test less flaky; it was reported that excessive scrolling may be the cause of Cypress failing to find elements
- _Describe the steps required to verify your changes are working as expected_
  > Thorough e2e testing locally
- _Describe the tests completed and the results_
  > Passing consistently
- _Exclusively stating 'Specs and automated tests passing' is NOT acceptable as appropriate testing
- _Optionally, provide a link to your [test plan](https://depo-platform-documentation.scrollhelp.site/developer-docs/create-a-test-plan-in-testrail) and [test execution records](https://depo-platform-documentation.scrollhelp.site/developer-docs/execute-tests-in-testrail)_

## Screenshots

N/A

## What areas of the site does it impact?

Decision review apps: HLR, NOD & Supplemental Claim

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [ ] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [ ] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

### :warning: Team Sites (only applies to modifications made to the VA.gov header) :warning:

- [ ] The vets-website header does not contain any web-components
- [ ] I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#local-dev) to test the injected header scenario
- [ ] I reached out in the `#sitewide-public-websites` Slack channel for questions

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: src/applications/appeals/shared/tests/cypress.setup.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+  cy.intercept('GET', '/v0/feature_toggles?*', mockFeatureToggles).as(
+    'features',
+  );
+  cy.intercept('GET', '/v0/maintenance_windows', []);
+  cy.intercept('GET', '/data/cms/vamc-ehr.json', mockVamc);
+
+  // contact page updates
+  cy.intercept('GET', '/v0/profile/status*', mockStatus);
+  // cy.intercept('GET', '/v0/profile/status/*', mockStatus);
```
Comment (jerekshoe) [inline src/applications/appeals/shared/tests/cypress.setup.js:26]: Should this comment be removed?

File: src/applications/appeals/shared/tests/cypress.helpers.js [hunk 2]
```diff
@@ -1,3 +1,136 @@
...
+            lastAction: 'field_grant',
+            date: null,
+          },
+        ],
+      },
+    },
+  ],
+};
+
+// export const mockCustomContestableIssues = ({ contestedIssues }) =>
```
Comment (jerekshoe) [inline src/applications/appeals/shared/tests/cypress.helpers.js:123]: Should this commented section be removed?

File: src/applications/appeals/996/tests/hlr.cypress.spec.js [hunk 3]
```diff
@@ -88,21 +92,16 @@ const testConfig = createTestConfig(
           });
         });
       },
+
+      'area-of-disagreement/:index': areaOfDisagreementPageHook,
+      // 'area-of-disagreement/0': areaOfDisagreementPageHook,
```
Comment (jerekshoe) [inline src/applications/appeals/996/tests/hlr.cypress.spec.js:97]: Should this comment be removed?

**Review by jerekshoe — APPROVED** [review]
Left a few comments about commented out code, but I didn't see anything that should block merging this as-is

File: src/applications/appeals/996/tests/hlr.cypress.spec.js (hunk 3 above)
Comment (Mottie) [reply src/applications/appeals/996/tests/hlr.cypress.spec.js]: Yes, done!

File: src/applications/appeals/shared/tests/cypress.helpers.js (hunk 2 above)
Comment (Mottie) [reply src/applications/appeals/shared/tests/cypress.helpers.js]: Yes, done!

**Pull request merged**